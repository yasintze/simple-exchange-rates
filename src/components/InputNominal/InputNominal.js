import React from "react";
import InputBase from "@material-ui/core/InputBase";
import { withStyles } from "@material-ui/core/styles";

type Props = {
  input: number,
  nominal: string,
  onFocus: Function,
  onBlur: Function,
  onChange: Function,
  classes: Object
};

const styles = theme => ({
  inputRoot: {
    color: "inherit",
    width: "100%"
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 120,
      "&:focus": {
        width: 200
      }
    }
  }
});

const InputNominal = (props: Props) => {
  const { input, nominal, onFocus, onBlur, onChange, classes } = props;
  return (
    <InputBase
      placeholder="Enter Nominal.."
      value={input}
      onChange={onChange}
      autoFocus
      inputProps={{
        onFocus,
        onBlur,
        "aria-label": "input-rate"
      }}
      name="input-rate"
      id="input-rate"
      classes={{
        root: classes.inputRoot,
        input: classes.inputInput
      }}
    />
  );
};

export default withStyles(styles)(InputNominal);
