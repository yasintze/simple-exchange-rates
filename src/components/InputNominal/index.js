import InputNominal from "./InputNominal";

export default InputNominal;
