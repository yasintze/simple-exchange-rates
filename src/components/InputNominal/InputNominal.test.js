// @flow
import React from "react";
import { shallow } from "enzyme";

import InputNominal from "./InputNominal";

describe("InputNominal component", () => {
  it("render correctly", () => {
    const wrapper = shallow(<InputNominal />);
    expect(wrapper).toMatchSnapshot();
  });
});
