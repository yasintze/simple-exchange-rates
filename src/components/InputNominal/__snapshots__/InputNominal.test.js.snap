// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNominal component render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(InputNominal) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "inputInput": "InputNominal-inputInput-2",
        "inputRoot": "InputNominal-inputRoot-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "inputInput": "InputNominal-inputInput-2",
          "inputRoot": "InputNominal-inputRoot-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
